### PR TITLE
fix(event): declare v1beta3 event paths under /api to match the served routes

### PR DESCRIPTION
**Pull request by fitzergerald**
## Summary

`schemas/constructs/v1beta3/event/api.yml` declared five event **write** operations without the `/api` prefix, while declaring five event **read** operations *with* it - in the same file. Meshery Cloud serves all of them under `/api`, so every generated client emitted URLs that no route answers.

The declared path key is the complete served path; nothing prepends `/api` at build time. So this failed only at runtime - the spec bundled, the generators ran, and the emitted client compiled and type-checked cleanly.

## User-visible consequence being fixed

Since **meshery-cloud v1.0.189 (2026-07-15)**, the browser's `create_session` audit POST has been hitting a 405. `RegisterOrgActivity` fires the mutation from a `useEffect` and never reads the result, so the failure produced no console error, no notification, and no retry.

**No event rows have been written since 2026-07-15 14:10 UTC.** That is the root cause of the Academy Instructors Console **ACTIVE LEARNERS** tile reading `0` for the current month. Ground truth is ~35-40 for the Meshery org and ~52-57 across all orgs holding learners.

The regression is [`ef1ec3f25bb`](https://github.com/layer5io/meshery-cloud/commit/ef1ec3f25bb) ("retire the 136 hand-authored RTK endpoints duplicating the generated cloudApi"), which moved `RegisterOrgActivity` from the hand-authored endpoint that posted to `/api/events` onto the generated `useCreateEventMutation` that posted to `/events`.

## Direction verified, not assumed

Against production (both probes unauthenticated, rejected before any handler, mutating nothing):

```
POST https://cloud.layer5.io/events      -> 405   # no such route
POST https://cloud.layer5.io/api/events  -> 401   # route exists, auth required
```

And against `meshery-cloud/server/router/router.go`, where `authedAPI`, `authByPasskeyAPI` and `authedORSpecialTokenAPI` are each `s.e.Group("/api")`. A second, independent corroboration: `meshery/meshery`'s `server/models/remote_provider.go:1893` hand-builds `%s/api/events/status` when calling the remote provider.

**The schema was the side that was wrong.** 405-vs-401 is the whole proof: the prefixed path is a real route, the unprefixed one is not.

## Is `/api` really the convention? Partly - and that is a finding

Across `schemas/constructs/*/*/api.yml`, effectively every path carries `/api`. The exceptions are exactly two constructs:

- **`event`** (v1beta1, v1beta2, v1beta3) - the defect fixed here.
- **`schedule`** (v1beta1, v1beta2) - **correct as-is, deliberately untouched.** meshery-cloud registers those routes on the bare router (`s.e.POST("/user/schedules", ...)`, router.go:521-535) outside the `/api` group, so the schema already matches what is served.

So `/api` is the convention for the authenticated API surface, but it is *not* a universal invariant. That distinction shaped the guard below.

## The duplicate-key trap, checked across all five paths

`/api/events` **already existed** as a path key carrying `getEventsAggregate`. Duplicate YAML mapping keys are not an error - one silently wins - so adding a second `/api/events:` would have dropped either the GET or the POST with no warning, generating cleanly and failing at runtime.

I checked all five corrected paths for the same collision. **Exactly one collides:**

| Declared path | Corrected path | Pre-existing key? |
|---|---|---|
| `/events` | `/api/events` | **YES - merged `post:` into the existing path item** |
| `/events/{eventId}` | `/api/events/{eventId}` | no |
| `/events/delete` | `/api/events/delete` | no |
| `/events/status` | `/api/events/status` | no |
| `/events/{eventId}/status` | `/api/events/{eventId}/status` | no |

Verified mechanically by parsing both revisions:

```
PRE-FIX  | duplicate-key error: none | path keys: 10 | unique: 10 | operations: 10
POST-FIX | duplicate-key error: none | path keys:  9 | unique:  9 | operations: 10
```

Path keys drop by exactly one (the merge); **all 10 operations are preserved**. Per-operation `x-internal` filtering still splits correctly: `cloud_openapi.yml` gets `/api/events` `{get, post}`, `meshery_openapi.yml` gets `{post}` only.

## `bulkDeleteEvents` - a second, separate defect (not fixed here)

The spec says `POST /api/events/delete`. The server implements `DELETE /api/events` **with a request body**, which `docs/http-api-design.md` explicitly forbids ("many HTTP clients and proxies strip it silently"). Here the **spec is the correct side and the server is out of step**, so only the prefix is corrected and the method is left alone.

Changing the spec to `DELETE /api/events` would entrench the forbidden shape. **Follow-up needed in meshery-cloud: add the `POST /api/events/delete` route.** This operation has no consumer today, so nothing is currently broken by the gap.

## Blast radius - every consumer enumerated and assessed

These operations are `x-internal: ["cloud", "meshery"]`, so they generate into **both** clients. All consumers were checked, not assumed.

| Consumer | Consumes generated event ops? | Impact |
|---|---|---|
| **meshery-cloud** | **Yes** - `ui/components/general/RegisterOrgActivity.tsx` uses `useCreateEventMutation` from `@meshery/schemas/cloudApi` | **Fixed by this PR.** The other four ops have no consumer - latent, not currently breaking anything. |
| **meshery/meshery** | **No** - verified, not repeated | Unaffected. Its `useDeleteEventMutation` is hand-authored in `ui/rtk-query/notificationCenter.ts` against `/api/system/events/*` (Meshery server's own surface), injected into its local `api`, not `@meshery/schemas/mesheryApi`. It imports `@meshery/schemas` only for types, enums and `meshkitError`. |
| **meshery-extensions** | **No** | Consumes `cloudApi`/`mesheryApi` (19 files) but no event operations. Unaffected. |
| **sistent** | Pass-through only | `src/schemas/index.tsx` re-exports `cloudApi`/`mesheryApi` wholesale; calls no event op. Carries the fix transparently. |
| layer5, layer5-academy, docs, meshery.io | No generated-client usage | Unaffected. |

**No consumer breaks.** The exported RTK hook surface is **byte-identical** in both clients before and after (verified by diffing the exported `use*Query`/`use*Mutation` symbol sets) - only URL string literals changed. No coordinated code change is required anywhere; meshery-cloud needs only the version pin bump.

Go models are unaffected: generation is models-only and embeds no paths.

## Recurrence guard

`make consumer-audit` already held **both halves** of this defect but reported them as unrelated counts:

```
Spec only (no handlers)     POST /events        <- what the schema declared
Handler only (no spec)      POST /api/events    <- what the server serves
```

Two lines, one defect, nothing correlating them. The audit now rejoins them and reports the pair as a single actionable finding naming the router file and line. Proven end-to-end against the pre-fix schema:

```
4 schema endpoints declared at a path no router serves, while the consumer serves
the same operation one "/api" prefix away:
  POST /events (operationId "createEvent", schemas/constructs/v1beta3/event/api.yml)
    has no handler, but meshery-cloud serves POST /api/events at
    server/router/router.go:587. The declared path is missing the "/api" prefix
    its router applies.
  ... (3 more)
```

It reports **0** against the fixed schema.

Two deliberate design choices:

- **Narrower than "every path must start with `/api`."** That blanket rule would wrongly flag `/user/schedules`. Only a schema path *with no handler* whose prefixed twin is *an orphaned handler* is reported - a correctly-matched unprefixed path can never trip it.
- **Tolerates path-parameter name drift** (`{eventId}` vs `{id}`), reusing the matcher's existing loose key. Without it, four of the five real cases slipped through on a naming difference that is not the defect. This was caught by running the guard against the actual pre-fix schema rather than trusting the unit tests.

Covered by 8 unit tests in `validation/prefix_mismatch_test.go`, including the `/user/schedules` false-positive guard and the inverse direction (schema carries a prefix the router does not apply).

## Documentation

- `docs/http-api-design.md` - new subsection "Declared paths are absolute - there is no implicit `/api` prefix": that the key is the full served path, how to read Echo group prefixes when verifying, how a prefix defect appears on both sides of the consumer audit, and that `/user/schedules` is a genuine exception to match rather than "fix".
- `AGENTS.md` (§ Casing rules at a glance) - the endpoint-path bullet now states the declared path is the full served path and that a missing prefix fails no build or validation step, linking to the above.

## Testing

| Command | Result |
|---|---|
| `make build` | pass (validate-schemas, bundle, Go, RTK, TS, enums, permissions, Go tests) |
| `make validate-schemas` | no blocking violations |
| `go test ./...` | pass |
| `npm run build` | pass |
| `make consumer-audit` | 0 prefix mismatches post-fix; 4 correctly reported pre-fix |

## Notes for the reviewer

- Generated artifacts are regenerated via `make build` and committed; nothing was hand-edited. `_openapi_build/` is gitignored.
- **v1beta1 and v1beta2 event are deliberately left alone.** Both are `x-superseded-by` (frozen published versions), and touching them would churn generated artifacts for contracts nobody should be on. They carry the same original defect.
- `validation/baseline/consumer-audit.txt` is deliberately not refreshed - it is a Phase 0 verbatim snapshot captured 2026-04-22, is not a CI gate, and has not tracked schema changes since.
- **Do not merge or release yet** pending the coordinated version-pin bump in meshery-cloud (`go.mod` and `ui/package.json` must move together - `TestSchemaAlignmentSuite` fails when they diverge, and the npm pin is exact on purpose).

## Summary by CodeRabbit

* **Bug Fixes**
  * Updated event API routes to consistently include the `/api` prefix.
  * Improved generated event clients to use the corrected routes.

* **New Features**
  * Added audit reporting for schema and consumer routes that differ only by the `/api` prefix.

* **Documentation**
  * Clarified complete API path requirements and added guidance for verifying route registrations.

* **Tests**
  * Added coverage for prefix mismatches, parameter variations, method mismatches, and related edge cases.

---

## Review round (commit 2)

Two findings raised, both engaged on merit:

- **`PrefixMismatch.String()` was direction-blind** (CodeRabbit; Copilot raised the same independently) - a **real defect in the guard this PR adds**. The detector deliberately handles both directions, but the diagnostic always said "missing the `/api` prefix", so the inverse case stated the opposite of the truth and would have told a contributor to add a prefix that was already one too many. Fixed with `SchemaOmitsPrefix()`, which reads direction off the consumer path (ground truth). Root cause worth naming: the inverse-direction test existed and passed, but asserted only the *pairing*, never the *message* - which is precisely how the wrong wording survived. Both directions now assert their diagnostic text.
- **AGENTS.md wording** - partially valid, taken with a narrower change than proposed. The bullet did lead with `/api` as a flat rule. But dropping the convention entirely (as proposed) invites the opposite error, since `/api` really is the convention for ~150 paths with one standing exception. The bullet now makes "the declared path must equal the route the server serves" governing, keeps `/api` as the convention, and names `/user/schedules` inline as the exception.

Both threads confirmed resolved by the reviewer.

The detector was re-verified end-to-end after the refactor. Worth flagging a trap in that verification: replaying the defect via `git show HEAD:...` is a silent no-op once the fix is committed, and "passes" by reporting zero. Replayed against `HEAD~1` it correctly reports all 4 with direction-correct wording, and 0 against the fixed schema.

### CODE REVIEW

File: validation/prefix_mismatch.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+// String renders a single actionable line for the audit report.
+func (p PrefixMismatch) String() string {
+	return fmt.Sprintf(
+		"%s %s (operationId %q, %s) has no handler, but %s serves %s %s at %s:%d. "+
+			"The declared path is missing the %q prefix its router applies.",
+		p.Method, p.SchemaPath, p.OperationID, p.SourceFile,
+		p.Repo, p.Method, p.ConsumerPath, p.RouterFile, p.RouterLine,
+		apiPrefix,
+	)
```
[the remark this answers is not in the document]
Comment (fitzergerald) [reply validation/prefix_mismatch.go:76]: Valid, and a real defect - fixed in b0918b65.

`String()` hardcoded the "missing prefix" remedy while `detectPrefixMismatches` deliberately handles both directions, so the inverse case reported the opposite of the truth and would have sent someone to add a prefix that was already one too many.

Added `SchemaOmitsPrefix()`, which reads the direction off the consumer path (ground truth - it is what the server actually serves), and `String()` now renders the matching remedy.

Worth naming the root cause: `TestDetectPrefixMismatches_SchemaCarriesPrefixRouterDoesNot` already exercised the inverse path, but only asserted the pairing and never the message - which is exactly why the wrong wording survived. Both directions now assert their diagnostic text.

File: AGENTS.md [hunk 2]
```diff
@@ -133,7 +133,7 @@ This repo IS the authority for that contract. The contract is not optional; devi
 - Object names: singular nouns (`model`, `component`, `design`)
 - `components/schemas` names: PascalCase nouns (`Model`, `Component`, `KeychainPayload`)
 - Files/folders: lowercase (`api.yml`, `keychain.yaml`, `templates/keychain_template.json`)
-- Endpoint paths: `/api` prefix, kebab-case, plural nouns (`/api/workspaces`, `/api/environments`)
+- Endpoint paths: `/api` prefix, kebab-case, plural nouns (`/api/workspaces`, `/api/environments`). The declared path is the **full served path** - nothing prepends `/api` at build time, so a missing prefix ships a client that 404s at runtime without failing any build or validation step. Verify against the consumer's route registration (Echo groups supply the prefix); see [`docs/http-api-design.md`](docs/http-api-design.md#declared-paths-are-absolute---there-is-no-implicit-api-prefix).
```
[the remark this answers is not in the document]
Comment (fitzergerald) [reply AGENTS.md]: Partially valid - taken, with a narrower change than proposed, in b0918b65.

The concern is right: the bullet led with `/api` as a flat rule, which could push someone to "fix" a correct bare route. But dropping the `/api` convention entirely invites the opposite error, since it genuinely is the convention for ~150 paths across the repo with exactly one standing exception.

So the bullet now makes "the declared path must equal the route the server serves" the governing rule, keeps `/api` as the convention for the authenticated surface, and names `/user/schedules` inline as the exception - rather than leaving the reader to find it in docs/http-api-design.md.

Worth noting the detector added in this PR already encodes that distinction: it is deliberately narrower than "every path must start with /api" precisely so a correct bare route can never be flagged.

File: validation/prefix_mismatch.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+}
```
[the remark this answers is not in the document]
Comment (fitzergerald) [reply validation/prefix_mismatch.go:77]: Already addressed in b0918b65 - this review raced that push (comment timestamped 16:24:16Z; the fix landed at 16:23). Same finding as the CodeRabbit thread above.

On the suggested derivation: I derive the direction from `hasAPIPrefix(ConsumerPath)` alone rather than comparing it against `hasAPIPrefix(SchemaPath)`. That is sufficient here because of how the pairing is constructed - candidates are built by either adding the prefix or removing it, never both, so a pair can only ever exist with the prefix on exactly one side. Comparing both would be redundant, and the consumer path is the side that is ground truth.
